api/app/src/sequelize/migrations/20230111000000-create-medical-models.ts

```diff
@@ -0,0 +1,97 @@
+import { DataTypes } from "sequelize";
+import type { Migration } from "..";
+import { Facility } from "../../models/medical/facility";
+import { Organization } from "../../models/medical/organization";
+import { Patient } from "../../models/medical/patient";
+import * as shared from "../migrations-shared";
+
+// Use 'Promise.all' when changes are independent of each other
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async transaction => {
+    await shared.createTable(queryInterface, transaction, Organization.NAME, {
+      id: {
+        type: DataTypes.STRING,
+        primaryKey: true,
+      },
+      cxId: {
+        type: DataTypes.UUID,
+        field: "cx_id",
+      },
+      organizationId: {
+        type: DataTypes.INTEGER,
+        field: "organization_id",
+        unique: true,
+      },
+      data: {
+        type: DataTypes.JSONB,
+      },
+    });
+
+    await shared.createTable(queryInterface, transaction, Facility.NAME, {
+      id: {
+        type: DataTypes.STRING,
+        primaryKey: true,
+      },
+      cxId: {
+        type: DataTypes.UUID,
+        field: "cx_id",
+      },
+      organizationId: {
+        type: DataTypes.INTEGER,
+        field: "organization_id",
+        references: { model: Organization.NAME, key: "organization_id" },
+      },
+      facilityId: {
+        type: DataTypes.INTEGER,
+        field: "facility_id",
+      },
+      data: {
+        type: DataTypes.JSONB,
+      },
+    });
+    // a raw query is necessary here as doing composite unique keys through sequelize doesn't work
+    await queryInterface.sequelize.query(
+      `ALTER TABLE facility ADD UNIQUE (organization_id, facility_id);`,
+      { transaction }
+    );
```

Comment:
That's weird, I just tried setting `uniqueKeys` on Sequelize `createTable`'s `options` param and it worked:

<img width="554" alt="image" src="https://user-images.githubusercontent.com/2132564/220215633-f6cfc574-2a52-45a5-8dba-c8c86d3b07f4.png">

<img width="562" alt="image" src="https://user-images.githubusercontent.com/2132564/220215678-2d558e79-72fd-4ed1-979b-042ee2937695.png">

I'd actually refactor our `createTable` so it gets `options` (`transaction`, `uniqueKeys`) instead of just adding a new parameter - something like this:

```ts
export type CreateTableOptions = Omit<QueryInterfaceCreateTableOptions, "transaction"> &
  DeepNonNullable<Required<Pick<QueryInterfaceCreateTableOptions, "transaction">>>;

export const createTable = async (
  queryInterface: QueryInterface,
  tableName: string,
  tableDefinitions: ModelAttributes,
  options: CreateTableOptions
) => {
  await queryInterface.createTable(
    tableName,
    {
      ...tableDefinitions,
      ...defaultColumnsDef(),
    },
    options
  );
  await addUpdatedAtTrigger(queryInterface, options.transaction, tableName);
};
```

Reply:
Interesting, didn't know that there was another way to do this. What I was doing was something like:

```
      organizationId: {
        type: DataTypes.INTEGER,
        field: "organization_id",
        unique: "some_key"
      },
      facilityId: {
        type: DataTypes.INTEGER,
        field: "facility_id",
        unique: "some_key"
      },
```

Which according to Sequelize: "If true, the column will get a unique constraint. If a string is provided, the column will be part of a composite unique index. If multiple columns have the same string, they will be part of the same unique index" -> however no index was being created

Will update accordingly

Reply:
Yeah, found somewhere on SO people talking about these approaches, and the `unique` "label" not working. 👍